### PR TITLE
Fix #2695 - Users Are Able To View Other User Detail By Entering ID Into URL Params

### LINKED ISSUE #2695: Other user's profiles accessable by non-admin users
Hi SuiteCRM team,

I encountered a potentially insecure behavior, which might be a bug.

#### Issue
Non-admin users can access other users' user detail and edit views by entering the respective URL with the user id. You can test this for example in the demo version:
Log in to http://demo.suiteondemand.com/ with the non-admin user 'will' / 'will'.
Open http://demo.suiteondemand.com/index.php?module=Users&action=EditView&record=1
You'll see the user module edit view for Joe Bloggs, the administrator. 
Even though you are not able to save changes, the other user records should not be accessible for non-admin users for confidentiality reasons. The user module's detail / edit views could show fields not part of the employee record in order to share information between the user and the admins. 

#### Expected Behavior
The user should see an access denied or be redirected to his/her own profile page

#### Possible Fix
I don't yet know, whether this should be fixed by code within the view.php files or whether it is due to a wrong configuration in the ACL default values.
The view.list.php has a mechanism to reject non-admin access, which could be implemented into the edit / detail view as well:

```
if ($_REQUEST['record'] != $GLOBALS['current_user']->id 
	&& !$GLOBALS['current_user']->isAdminForModule('Users')
	&& !$GLOBALS['current_user']->isDeveloperForModule('Users')
 ){
         //instead of just dying here with unauthorized access will send the user back to his/her settings
         SugarApplication::redirect('index.php?module=Users&action=EditView&record=' . $GLOBALS['current_user']->id);
 }
```

#### Your Environment
Tested on the latest suiteCRM version 7.7.6 and the demo site.

### PR DESCRIPTION
**Pull request by salesagility-davidthomson**
## Description
reference #2695 
Added check into users view.edit.php and view.detail.php to check if user is allow to edit or view requested userid if not redirect to index 

## Motivation and Context
Security bug

## How To Test This
Test if users admin can still view all users
Test that non admin logged in user cannot view another users details and that the user is indeed redirected to index
Test that a user can change password from forgot password section

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/Users/views/view.edit.php [hunk 1]
```diff
@@ -44,9 +44,18 @@
...
- 	function __construct(){
- 		parent::__construct();
- 	}
+    function __construct(){
+        parent::__construct();
+
+        $recordid = $_REQUEST['record'];
+        $currentuserid = $GLOBALS['current_user']->id;
+        $isAdmin = $GLOBALS['current_user']->isAdminForModule('Users');
+        if( $recordid == $currentuserid || $isAdmin) {
```
Comment (samus-aran) [inline modules/Users/views/view.edit.php]: Would it not be better to have a negative?

File: modules/Users/views/view.detail.php [hunk 2]
```diff
@@ -43,9 +43,9 @@
 
 class UsersViewDetail extends ViewDetail {
 
- 	public function __construct(){
- 		parent::__construct();
- 	}
+    function __construct(){
```
Comment (samus-aran) [inline modules/Users/views/view.detail.php]: redeclared this as public so there isn't any ambiguity

File: modules/Users/views/view.edit.php [hunk 3]
```diff
@@ -44,9 +44,9 @@
 
 class UsersViewEdit extends ViewEdit {
 var $useForSubpanel = true;
- 	function __construct(){
- 		parent::__construct();
- 	}
+    function __construct(){
```
Comment (samus-aran) [inline modules/Users/views/view.edit.php]: redeclared this as public so there isn't any ambiguity

**Comment by salesagility-davidthomson** [comment]
I've included some files that should not have been committed. I'm going to close this and recreate it with only the right files.

**Pull request closed without merging**